assets/javascript/app.js

```diff
@@ -0,0 +1,154 @@
+            answer: 2
+        }, 
+        {
+            question: "How many Infinity Stones are there in the Marvel Universe?", 
+            choice: ["12", "5", "9", "6"],
+            answer: 3
+        }, 
+        {
+            question: "In the show,'Family Guy', what is the name of the Griffin's dog?", 
+            choice: ["Scooby", "Brian", "Bryan", "Marmaduke" ],
+            answer: 1
+        }, 
+        {
+            question: "What is Goku's race?", 
+            choice: ["Namekian", "Saiyan", "Human", "Ape"],
+            answer: 1
+        }, 
+        {
+            question: "What element can Magneto control?", 
+            choice: ["Metal", "Fire", "Water", "Air"],
+            answer: 0
+        }];
+    
+    var correctCount = 0;
+    var wrongCount = 0;
+    var unanswerCount = 0;
+    var timer = 20;
+    var intervalId;
+    var userGuess ="";
+    var running = false;
+    var qCount = options.length;
+    var pick;
+    var index;
+    var newArray = [];
+    var holder = [];
+    
+    $("#reset").hide();
+    
+    $("#start").on("click", function () {
+            $("#start").hide();
+            displayQuestion();
+            runTimer();
+            for(var i = 0; i < options.length; i++) {
+        holder.push(options[i]);
+    }
+        })
+
+    function runTimer(){
+        if (!running) {
+        intervalId = setInterval(decrement, 1000); 
+        running = true;
+        }
+    }
+
+    function decrement() {
+        $("#timer").html("<h3>Time remaining: " + timer + "</h3>");
+        timer --;
+        if (timer === 0) {
+            unanswerCount++;
+            stop();
+            $("#answers").html("<p>Time is up! The correct answer is: " + pick.choice[pick.answer] + "</p>");
+            results();
+        }	
+    }
+    
+    function stop() {
+        running = false;
+        clearInterval(intervalId);
+    }
+
+    function displayQuestion() {
+        index = Math.floor(Math.random()*options.length);
+        pick = options[index];
+            $("#questions").html("<h2>" + pick.question + "</h2>");
+            for(var i = 0; i < pick.choice.length; i++) {
+                var userChoice = $("<div>");
+                userChoice.addClass("answerchoice");
+                userChoice.html(pick.choice[i]);
+                userChoice.attr("data-guessvalue", i);
+                $("#answers").append(userChoice);
+    }
+    
+    $(".answerchoice").on("click", function () {
+        userGuess = parseInt($(this).attr("data-guessvalue"));
+        if (userGuess === pick.answer) {
+            stop();
+            correctCount++;
+            userGuess="";
+            $("#answers").html("<p>Correct!</p>");
+            results();
+        } else {
+            stop();
+            wrongCount++;
+            userGuess="";
+            $("#answers").html("<p>Wrong! The correct answer is: " + pick.choice[pick.answer] + "</p>");
+            results();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['results' was used before it was defined. (no-use-before-define)](https://app.codacy.com/app/columbia-fsf-f2019/trivia-game/pullRequest?prid=4555496)